src/gpio.rs

```diff
@@ -14,7 +14,7 @@ pub trait GpioExt {
     type Parts;
 
     /// Splits the GPIO block into independent pins and registers
-    fn split(self, rcc: &mut Rcc) -> Self::Parts;
+    fn split(self, rcc: &Rcc) -> Self::Parts;
```

Comment:
Also, unrelated to #180, since `Enable::enable` and `Reset::reset` use bit banding, we do not need exclusive access to the rcc right?

Reply:
Will remove this and perhaps do that later